src/ring_theory/subsemiring/basic.lean

```diff
@@ -415,7 +415,32 @@ instance {R} [semiring R] : comm_semiring (center R) :=
 { ..submonoid.center.comm_monoid,
   ..(center R).to_semiring}
 
-end
+end center
+
+section centralizer
+
+/-- The centralizer of a set as subsemiring. -/
+def centralizer {R} [semiring R] (s : set R) : subsemiring R :=
+{ carrier := s.centralizer,
+  zero_mem' := by { intros y h, simp, },
+  add_mem' := λ a b ha hb c mc, by rw [add_mul, mul_add, ha _ mc, hb _ mc],
```

Comment:
Can you extract this to `set.zero_mem_centralizer` and `set.add_mem_centralizer` lemmas to match `set.one_mem_centralizer` and `set.mul_mem_centralizer`? I imagine you can weaken the assumptions from `semiring` to something weaker on those lemmas.